### PR TITLE
Added Assignment::Creator class

### PR DESCRIPTION
**Pull request by EricPickup**
## What
This PR adds a `Creator` class for `Assignment` so that we can isolate everything related to assignment creation outside of the controller. We'll be adding a lot more logic to assignment creation with #2241.

We already have an `Assignment::Editor` class and this is similar to `Organization::Creator`, `Group::Creator`, and `Roster::Creator`.

## Notes
- This is under a feature flag and will be gradually rolled out
- Old logic/tests inside the controller will be removed once we're 100% with this Creator
- I plan to merge group assignment creation into this creator, unifying `GroupAssignment` and `Assignment` creation under one creator class

### CODE REVIEW

File: spec/models/assignment/creator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      end
+
+      it "returns result containing unsaved assignment" do
+        result = subject.perform(options: options)
+        expect(result.assignment.persisted?).to be false
+      end
+    end
+  end
+
+  context "deadlines" do
```
Comment (EricPickup) [inline spec/models/assignment/creator_spec.rb]: All of these deadline tests are the exact same tests [taken from `AssignmentsController`](https://github.com/education/classroom/blob/master/spec/controllers/assignments_controller_spec.rb#L133). The old tests in the controller will be removed once we get rid of the feature flag.

**Comment by thebradbain** [comment]
I don't have too much experience with the `Creator` pattern, but is there a plan to also create a `GroupAssignment::Creator` class? Or will #2241 not concern `GroupAssignmentRepo` and `GroupAssignment`?

**Comment by EricPickup** [comment]
@thebradbain I'm hoping to merge GroupAssignment creation into this same creator after this PR goes in. `GroupAssignment` already shares `Assignment::Creator` for editing assignments

File: app/models/assignment/creator.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class Assignment
+  class Creator
+    attr_reader :assignment, :options
+
+    class Result
+      class Error < StandardError; end
```
Comment (jeffrafter) [inline app/models/assignment/creator.rb]: I don't see this being used...

**Pull request closed without merging**